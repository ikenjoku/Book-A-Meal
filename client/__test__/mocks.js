
export const meals = [
  {
    id: 0,
    name: 'Nigerian Jollof',
    description: 'Max Jollof rice, plantain, grilled turkey, bellefull Coke',
    price: 1100,
    imageurl: 'https://africa-publicmages/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 1,
    name: 'Rice and Beans',
    description: 'Rice, Beans, Plantain, Panla Sauce, Max Coke',
    price: 1500,
    imageurl: 'https://africa-g/images/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 2,
    name: 'Egusi Semo',
    description: 'Semolina, Egusi Soup, Meat, Extra',
    price: 1200,
    imageurl: 'https://africa-public.food.jcts/80/80418_1465475724_ma.jpg',
  },
  {
    id: 3,
    name: 'Green Calulu',
    description: 'Chicken salad with extra cabbage',
    price: 1250,
    imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 4,
    name: 'Onion Jappatti',
    description: 'Ingera palate with raw meat sauce',
    price: 2200,
    imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
];

export const testMeal = {
  id: 3234,
  name: 'Strawberry flakes',
  description: 'Ingera palate with raw meat sauce',
  price: 2200,
  imageurl: 'https://dynamic/prodg/images/produc475724_ma.jpg',
  createdAt: new Date(),
  updatedAt: new Date(),
};

export const mockStoreData = {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [
      {
        id: 3,
        name: 'Green Calulu',
        description: 'Chicken salad with extra cabbage',
        price: 1250,
        imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
      {
        id: 4,
        name: 'Onion Jappatti',
        description: 'Ingera palate with raw meat sauce',
        price: 2200,
        imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
    ],
    error: null,
    isLoading: false,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders: false,
  },
};

export const createdMenu = {
  message: 'Menu for 2018-06-15 was successfully created',
  newMenu: {
    id: 24,
    date: '2018-06-15',
    updatedAt: '2018-07-02T17:37:46.023Z',
    createdAt: '2018-07-02T17:37:46.023Z',
  },
};

export const menus = [{
  id: 23,
  date: '2018-07-02',
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
},
{
  id: 24,
  date: new Date().toISOString().substr(0, 10),
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
}];

export const orders = [
  {
    id: 2,
    date: '2018-05-16',
    amount: 1500,
    userId: 1,
    mealId: 2,
    status: 'cancelled',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 1,
      firstname: 'Ezinne',
      lastname: 'Njoku',
    },
    Meal: {
      id: 2,
      name: 'Rice and Beans',
      price: 1500,
    },
  },
  {
    id: 3,
    date: '2018-05-16',
    amount: 1200,
    userId: 2,
    mealId: 3,
    status: 'pending',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 2,
      firstname: 'Kelechi',
      lastname: 'Njoku',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
  {
    id: 4,
    date: '2018-05-16',
    amount: 1500,
    userId: 3,
    mealId: 3,
    status: 'delivered',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 3,
      firstname: 'Nnamdi',
      lastname: 'James',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
];
