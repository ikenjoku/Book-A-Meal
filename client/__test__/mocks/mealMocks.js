export const getAllMealsMock = [
  {
    name: 'test meal',
    price: 1230,
    description: 'this is a very nice meal',
    imageurl: 'https://sdfsdjhbsn/sdf.asp',
  },
];
