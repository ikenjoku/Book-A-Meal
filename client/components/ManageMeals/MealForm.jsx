import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

class MealForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      meal: {
        name: props.meal ? props.meal.name : '',
        description: props.meal ? props.meal.description : '',
        price: props.meal ? props.meal.price : '',
        imageurl: props.meal ? props.meal.imageurl : '',
      },
      error: '',
    };
  }

  onFormInput = (event) => {
    event.preventDefault();
    const meal = this.state.meal;
    switch (event.target.name) {
      case 'imageurl':
        meal.imageurl = event.target.files[0];
        break;
      default:
        meal[event.target.name] = event.target.value;
    }
    this.setState({ meal });
  }

  onFormSubmit = (event) => {
    event.preventDefault();
    const { name, price, description, imageurl } = this.state.meal;
    if (name && price && description && imageurl) {
      this.setState(() => ({ error: '' }));

      const formData = new FormData();
      formData.append('name', name);
      formData.append('price', price);
      formData.append('description', description);
      formData.append('imageurl', imageurl);
      this.props.onSubmit(formData);
    } else {
      this.setState(() => ({ error: 'Please fill all the fields' }));
    }
  }

  render() {
    return (
      <div>
        <form
          id="meal-form"
          onSubmit={this.onFormSubmit}
          name="mealForm"
          encType="multipart/form-data"
          className="add-meal-form"
        >
          {this.state.error &&
          <div className="alert alert-danger">
            {this.state.error}
          </div>}

          <input
            className="form-control"
            type="text"
            name="name"
            value={this.state.meal.name}
            onChange={this.onFormInput}
            placeholder="Meal Name"
          />

          <input
            className="form-control"
            type="text"
            name="description"
            value={this.state.meal.description}
            onChange={this.onFormInput}
            placeholder="Short Description"
          />

          <input
            className="form-control"
            type="number"
            name="price"
            value={this.state.meal.price}
            onChange={this.onFormInput}
            placeholder="Price"
          />

          <input
            className="form-control"
            type="file"
            name="imageurl"
            onChange={this.onFormInput}
          />

          <div className="centerBtn">
            <input
              className="mealFormBtn"
              type="submit"
              value="Submit"
            />

            <Link
              to="/meals"
            >
              <div className="mealFormBtn">Go Back</div>
            </Link>
          </div>
        </form>
      </div>
    );
  }
}

MealForm.propTypes = {
  meal: PropTypes.object,
  onSubmit: PropTypes.func.isRequired,
};

MealForm.defaultProps = {
  meal: undefined,
};

export default MealForm;
