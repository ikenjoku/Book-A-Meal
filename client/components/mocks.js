

export const sampleOrders = {
  message: 'All orders retrieved successfully',
  orders: [
    {
      id: 2,
      date: '2018-05-16',
      amount: 1500,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 3,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 4,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 1,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 5,
      date: '2018-05-23',
      amount: 1500,
      userId: 2,
      mealId: 2,
      status: 'pending',
      createdAt: '2018-05-23T23:55:05.091Z',
      updatedAt: '2018-05-23T23:55:05.091Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 7,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 6,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 8,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
  ],
};

