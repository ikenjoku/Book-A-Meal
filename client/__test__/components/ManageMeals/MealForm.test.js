import React from 'react';
import { shallow } from 'enzyme';
import MealForm from '../../../components/ManageMeals/MealForm';
import { meals } from '../../mocks';

describe('MealForm Component', () => {
  let wrapper, onSubmit;
  onSubmit = jest.fn();
  test('it should render the MealForm correctly', () => {
    wrapper = shallow(<MealForm onSubmit={onSubmit}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('it should render the MealForm with data correctly', () => {
    wrapper = shallow(<MealForm meal={meals[1]}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onFormInput when input fields change value', () => {
    const wrapper = shallow(<MealForm onSubmit={onSubmit} />);
    const onFormInputSpy = jest.spyOn(
      wrapper.instance(), 'onFormInput'
    );
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'Eja nla',
        name: 'name'
      }
    };
    wrapper.instance().onFormInput(event);
    expect(onFormInputSpy).toHaveBeenCalledTimes(1);
  });

  it('should call onFormInput when file is uploaded', () => {
    const wrapper = shallow(<MealForm onSubmit={onSubmit} />);
    const onFormInputSpy = jest.spyOn(
      wrapper.instance(), 'onFormInput'
    );
    const event = {
      preventDefault: jest.fn(),
      target: {
        files: [{name: 'mypix.jpg'}],
        name: 'imageurl'
      }
    };
    wrapper.instance().onFormInput(event);
    expect(onFormInputSpy).toHaveBeenCalledTimes(1);
  });

  it('should call onFormSubmit when form is submited with valid complete data', () => {
    const wrapper = shallow(<MealForm onSubmit={onSubmit} meal={meals[1]} />);
    const onFormSubmitSpy = jest.spyOn(
      wrapper.instance(), 'onFormSubmit'
    );
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.instance().onFormSubmit(event);
    expect(onFormSubmitSpy).toHaveBeenCalledTimes(1);
  });

  it('should set error when onFormSubmit is called with invalid data', () => {
    const wrapper = shallow(<MealForm onSubmit={onSubmit} meal={{name: 'Fish Nose'}} />);
    const onFormSubmitSpy = jest.spyOn(
      wrapper.instance(), 'onFormSubmit'
    );
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.instance().onFormSubmit(event);
    expect(onFormSubmitSpy).toHaveBeenCalledTimes(1);
  });
});
