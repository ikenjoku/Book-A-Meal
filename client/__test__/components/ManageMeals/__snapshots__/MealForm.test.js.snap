// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the MealForm correctly 1`] = `
<div>
  <form
    className="add-meal-form"
    encType="multipart/form-data"
    name="mealForm"
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="name"
      onChange={[Function]}
      placeholder="Meal Name"
      type="text"
      value=""
    />
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Short Description"
      type="text"
      value=""
    />
    <input
      className="form-control"
      name="price"
      onChange={[Function]}
      placeholder="Price"
      type="number"
      value=""
    />
    <input
      className="form-control"
      name="imageurl"
      onChange={[Function]}
      type="file"
    />
    <div
      className="centerBtn"
    >
      <input
        className="mealFormBtn"
        type="submit"
        value="Submit"
      />
      <Link
        replace={false}
        to="/meals"
      >
        <div
          className="mealFormBtn"
        >
          Go Back
        </div>
      </Link>
    </div>
  </form>
</div>
`;

exports[`it should render the MealForm with data correctly 1`] = `
<div>
  <form
    className="add-meal-form"
    encType="multipart/form-data"
    name="mealForm"
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="name"
      onChange={[Function]}
      placeholder="Meal Name"
      type="text"
      value="Rice and Beans"
    />
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Short Description"
      type="text"
      value="Rice, Beans, Plantain, Panla Sauce, Max Coke"
    />
    <input
      className="form-control"
      name="price"
      onChange={[Function]}
      placeholder="Price"
      type="number"
      value={1500}
    />
    <input
      className="form-control"
      name="imageurl"
      onChange={[Function]}
      type="file"
    />
    <div
      className="centerBtn"
    >
      <input
        className="mealFormBtn"
        type="submit"
        value="Submit"
      />
      <Link
        replace={false}
        to="/meals"
      >
        <div
          className="mealFormBtn"
        >
          Go Back
        </div>
      </Link>
    </div>
  </form>
</div>
`;
